src/components/core/NotifyObservers.react.js

```diff
@@ -17,79 +17,49 @@ function mapStateToProps(state) {
 }
 
 function mapDispatchToProps(dispatch) {
-    return {dispatch};
+    return { dispatch };
 }
 
 function mergeProps(stateProps, dispatchProps, ownProps) {
-    const {dispatch} = dispatchProps;
+    const { dispatch } = dispatchProps;
+
     return {
         id: ownProps.id,
         children: ownProps.children,
         dependencies: stateProps.dependencies,
         paths: stateProps.paths,
 
         setProps: function setProps(newProps) {
-            const payload = {
+            // Identify the modified props that are required for callbacks
+            const watchedKeys = filter(key =>
+                stateProps.dependencies &&
+                stateProps.dependencies.find(dependency =>
+                    dependency.inputs.find(input => input.id === ownProps.id && input.property === key) ||
+                    dependency.state.find(state => state.id === ownProps.id && state.property === key)
+                )
+            )(keysIn(newProps));
+
+            // Always update this component's props
+            dispatch(updateProps({
                 props: newProps,
                 id: ownProps.id,
-                itempath: stateProps.paths[ownProps.id],
-            };
+                itempath: stateProps.paths[ownProps.id]
+            }));
```

Comment:
The component used to only be updated when it had a watched prop -- this makes no sense as it prevents components from updating themselves and have consistent behavior in the FE in all usage scenarios.